fastlane/lib/fastlane/fast_file.rb

```diff
@@ -309,12 +308,16 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile', version
           if Dir[clone_folder].empty?
             UI.message("Cloning remote git repo...")
             Helper.with_env_values('GIT_TERMINAL_PROMPT' => '0') do
+              command = "git clone #{url.shellescape} #{clone_folder.shellescape} --no-checkout"
               # When using cached clones, we need the entire repository history
               # so we can switch between tags or branches instantly, or else,
               # it would defeat the caching's purpose.
-              depth = is_eligible_for_caching ? "" : "--depth 1"
-
-              Actions.sh("git clone #{url.shellescape} #{clone_folder.shellescape} #{depth} --no-checkout #{branch_option}")
+              command << " --depth 1" unless is_eligible_for_caching
+              command << " --branch #{branch}" if branch != 'HEAD'
+              git_extra_headers.each do |header|
+                command << " --config \"http.extraHeader=#{header}\""
+              end
+              Actions.sh(command)
```

Comment:
We need to also `shellescape` the `#{branch}` name, and typically the extra headers as well.
Otherwise, not only is there a risk that the command would not work if the branch name or extra headers contain spaces or quotes, but this would also open the action to (accidental or intentional) command injection.

In fact, instead of calling `.shellescape` on each parameter ourselves (with the risk of forgetting one of them in the process), I always recommend to pass arguments separately to `Actions.sh` / `Open3.popen` / `system` / …,  —as opposed to passing them as a single string—so that the string don't even have to go through shell argument parsing (i.e. splitting the string in arguments, interpreting quotes and backslashes, solving env vars…) and instead the already-isolated arguments are directly passed independently to the process being called, without risking being (mis-)interpreted by the shell, and without needing to `.shellescape` them:

```suggestion
              command = ['git', 'clone', url, clone_folder, '--no-checkout']
              command += ['--depth', '1'] unless is_eligible_for_caching
              command += ['--branch', branch] unless branch == 'HEAD'
              git_extra_headers.each do |header|
                command += ['--config', "http.extraHeader=#{header}"]
              end
              Actions.sh(*command)
```

Reply:
Sounds and looks reasonable to me (I'm no expert on command injection in Ruby). 

But just to make sure I understand you correctly: The "issue" that you are fixing here is not something that was introduced with this PR, right? It was already there to begin with (for e.g. the `branch` parameter).

Reply:
Good call @jenskuhrjorgensen! I would handle this in a separate PR, while also fixing all the `Actions.sh` calls, not only this one.

WDYT @AliSoftware?

Reply:
Correct, this was already an issue with the previous implementation injecting `#{branch}` without escaping it.

Though I think that, while it's unusual to have a branch name with spaces or special characters to begin with, it'd be much more probable to have a header with spaces and/or special characters. For example, if your Bearer token you want to provide as the `Authorization` header contains `$` somewhere in the token, the shell would interpret it as an env var interpolation. Similar issues with many other special characters like quotes, etc that could easily appear in custom headers (more likely than them appearing in git branch names)

Reply:
> I would handle this in a separate PR, while also fixing all the Actions.sh calls, not only this one.
> WDYT @AliSoftware?

@revolter I don't think that's the best way to go, because:
 - While I definitively agree it would be really great to fix all the occurrences of that mistake (of passing a string instead of separate parameters to `sh`) all over the _fastlane_ codebase…
 - … in practice there potentially a lot of those occurrences to review.
 - Which might thus make handling them all in a single follow-up PR… become quite a large task, especially if you also include the work needed to update all the unit tests to match
 - And as a result, I fear that this will lead to this follow-up PR never being created/reviewed/merged anytime soon. Especially given that many of us contributors have limited bandwidth which makes some of those cleanup/refactoring tasks being postponed forever for lack of available time to spend on them 😅 

So personally I'd rather recommend an iterative approach, i.e. solving this each time we stumble upon a case in the codebase where we did it incorrectly (single string instead of parameter list).

Then if someone has the bandwidth to address all the other occurrences of this elsewhere in the fastlane codebase, they're definitively welcome to do a separate PR to fix all the implementations + all the unit tests across the codebase at once.

But I don't want to risk such a follow-up PR to never be done, especially given the case of special characters in custom headers (like `Authorization: Bearer a2z6$pXn`—note the `$`—, `If-None-Match: "c561c68"`, `E-Tag: "12345"`, `Cookie: user=John "Apple" Appleseed`—note the quotes) might be more common, and thus actually cause not just security issues but also actual bugs in this `import_from_git` action specifically.

Reply:
> all over the fastlane codebase…

Sorry, I was not clear 🙈 I was thinking of fixing them only in the `import_from_git` function.